apps/api/e2e/tests/leaderboard.test.ts

```diff
@@ -80,12 +81,35 @@ describe("Leaderboard API", () => {
       reason: "Test trade",
     });
 
+    await adminClient.endCompetition(startResponse.competition.id);
```

Comment:
All leaderboard tests were refactored to include:
- all competitions are finished (because data from in-flight competitions don't make it to the global leaderboard anymore)
- a fake update on agent rank ( this will be removed later when we have the agent rank update logic)
- the check for active agents